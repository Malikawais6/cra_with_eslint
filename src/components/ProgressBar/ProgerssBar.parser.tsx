export const getBarColor = (percent: number) => {
  if (percent < 40) {
    return "#ce4c00";
  } else if (percent >= 40 && percent <= 70) {
    return "#0000FF";
  } else {
    return "#00c972";
  }
};
