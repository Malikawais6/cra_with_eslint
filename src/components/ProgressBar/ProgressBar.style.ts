import styled from "styled-components";
import { Progress } from "antd";

export const ProgressStyled = styled(Progress)`
  .ant-progress-bg {
    height: 14px !important;
  }
`;

export const ProgressBarStory = styled.div`
  width: 300px;
`;
