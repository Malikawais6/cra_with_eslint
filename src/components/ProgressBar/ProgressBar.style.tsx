import styled from "styled-components";

export const ProgressStyled = styled.div`
  width: 300px;
  margin-left: 40px;
  .ant-progress-bg {
    height: 14px !important;
  }
`;
