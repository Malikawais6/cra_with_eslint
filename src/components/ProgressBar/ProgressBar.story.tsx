import React from "react";
import { storiesOf } from "@storybook/react";

import { ProgressBar } from "./ProgressBar";

storiesOf("Progress Bar", module).add("Progress Bar Default", () => (
  <ProgressBar percent={50} />
));

storiesOf("Progress Bar", module).add("Progress Bar Success", () => (
  <ProgressBar percent={100} status="success" />
));

storiesOf("Progress Bar", module).add("Progress Bar Failure", () => (
  <ProgressBar percent={30} status="exception" />
));
