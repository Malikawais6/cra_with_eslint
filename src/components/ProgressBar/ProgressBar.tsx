import React from "react";
import { Progress } from "antd";
import "antd/dist/antd.css";
import { ProgressProps } from "antd/lib/progress/";

import { ProgressStyled } from "./ProgressBar.style";

export const ProgressBar = (props: ProgressProps) => {
  return (
    <ProgressStyled>
      <Progress
        percent={props.percent}
        status={props.status ? props.status : "normal"}
        showInfo={false}
      ></Progress>
    </ProgressStyled>
  );
};
