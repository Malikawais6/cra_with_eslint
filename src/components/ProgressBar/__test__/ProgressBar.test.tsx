import * as React from "react";
import { render } from "@testing-library/react";

import { ProgressBar } from "../ProgressBar";

function renderField() {
  return <ProgressBar />;
}

it("Make sure Progress Bar renders properly", () => {
  const { container } = render(renderField());
  const ProgressBarElement = container.getElementsByClassName("ant-progress");

  expect(ProgressBarElement.length).not.toBeLessThan(1);
});
