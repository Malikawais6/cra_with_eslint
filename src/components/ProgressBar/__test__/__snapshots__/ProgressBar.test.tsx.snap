// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make sure Progress Bar renders properly 1`] = `
<div
  class="sc-bdVaJa kxWhAr"
>
  <div
    class="ant-progress ant-progress-line ant-progress-status-normal ant-progress-default"
  >
    <div>
      <div
        class="ant-progress-outer"
      >
        <div
          class="ant-progress-inner"
        >
          <div
            class="ant-progress-bg"
            style="width: 0%; height: 8px;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
